### ISSUE TITLE
More Bedrock Model Support

### ISSUE THREAD

**Issue by ionflow** [Mirascope/mirascope#319]
### Description

Assuming AWS has better uptime percentage, it would be good to support more of their models for enterprise customers. Specifically I'm interested in the following models:

Claude (latest)
```
anthropic.claude-3-opus-20240229-v1:0
anthropic.claude-3-sonnet-20240229-v1:0
anthropic.claude-3-haiku-20240307-v1:0
```
Meta (latest)
```
meta.llama3-70b-instruct-v1:0
meta.llama3-8b-instruct-v1:0
```
Mistral (latest)
```
mistral.mistral-large-2402-v1:0
mistral.mixtral-8x7b-instruct-v0:1
mistral.mistral-7b-instruct-v0:2
```
Cohere (latest)
```
cohere.command-r-plus-v1:0
cohere.command-r-v1:0
```

**Comment by willbakst** [comment]
In v1, we've added support for LiteLLM, which should enable all of the models on Bedrock. Is this sufficient? Or are you looking more for direct support @ionflow?

**Comment by agentgill** [comment]
Ability to invoke models on AWS would be useful. I guess it some sort of wrapper around boto3 bedrock-runtime

**Comment by willbakst** [comment]
Yeah I think LiteLLM has comparable support, which we've added support for in v1.

I'm not 100% sure what our own version for direct support would look like, but I'm happy to look into this if the LiteLLM support is insufficient for whatever reason.

**Comment by willbakst** [comment]
Looks like we should be able to use the BedrockRuntime `converse` and `converse_stream` methods to implement support for `bedrock.call` so we can access Bedrock more directly (rather than using something like LiteLLM).

This is a great first issue as there are already 9 other providers implemented that can act as a good reference for how one should implement this using our internal helper methods and factories.

We also have documentation for adding support for [Custom LLM Providers](https://docs.mirascope.io/latest/integrations/custom_provider/), which is the same process for adding support as is required for adding Bedrock support.

Converse: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock-runtime/client/converse.html
Converse Stream: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock-runtime/client/converse_stream.html

**Comment by willbakst** [comment]
Released in `v1.4`!
